src/Network/PeerDiscovery/Util.hs

```diff
@@ -92,6 +96,20 @@ modifyMVarP mv f = modifyMVar mv $ \v -> do
 modifyMVarP_ :: MVar a -> (a -> a) -> IO ()
 modifyMVarP_ mv f = modifyMVar_ mv $ \v -> return $! f v
 
+-- | Strictly apply a pure function to the contents of a 'TVar'
+-- and return a result.
+modifyTVarP :: TVar a -> (a -> (a, r)) -> STM r
+modifyTVarP tv f = do
```

Comment:
This one is unused. Let's only bother if we need it. There's already a `modifyTVar` in `Control.Concurrent.STM.TVar`.